mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageVaultHookTest.java

```diff
@@ -0,0 +1,224 @@
+
+    private static final Instant IN_SLICE = Instant.parse("2019-02-27T14:15:30.00Z");
+    private static final MailboxPath MAILBOX_ALICE_ONE = MailboxPath.forUser(ALICE_ADDRESS, "ALICE_ONE");
+    private static final MailboxPath MAILBOX_BOB_ONE = MailboxPath.forUser(BOB_ADDRESS, "BOB_ONE");
+
+    private MailboxManager mailboxManager;
+    private MessageIdManager messageIdManager;
+    private Message mailContent;
+    private MemoryDeletedMessagesVault messageVault;
+    private Date internalDate;
+    private Clock clock;
+    private SessionProvider sessionProvider;
+    private MailboxSession aliceSession;
+    private MailboxSession bobSession;
+    private InMemoryMailboxSessionMapperFactory mapperFactory;
+    private CombinationManagerTestSystem testingData;
+    private SearchQuery searchQuery;
+
+    private DeletedMessage buildDeletedMessage(List<MailboxId> mailboxIds, MessageId messageId, User user) throws Exception {
+        return DeletedMessage.builder()
+            .messageId(messageId)
+            .originMailboxes(mailboxIds)
+            .user(user)
+            .deliveryDate(ZonedDateTime.ofInstant(internalDate.toInstant(), ZoneOffset.UTC))
+            .deletionDate(ZonedDateTime.ofInstant(clock.instant(), ZoneOffset.UTC))
+            .sender(MaybeSender.getMailSender(ALICE_ADDRESS))
+            .recipients(new MailAddress(TEST_ADDRESS))
+            .hasAttachment(false)
+            .subject("test")
+            .build();
+    }
+
+    private ComposedMessageId appendMessage(MessageManager messageManager) throws Exception {
+        return messageManager.appendMessage(MessageManager.AppendCommand.builder()
+                .withInternalDate(internalDate)
+                .build(mailContent), aliceSession);
+    }
+
+    @BeforeEach
+    void setUp() throws Exception {
+        clock = Clock.fixed(IN_SLICE, ZoneOffset.UTC);
+        messageVault = new MemoryDeletedMessagesVault();
+
+        Authenticator noAuthenticator = null;
+        Authorizator noAuthorizator = null;
+        sessionProvider = new SessionProvider(noAuthenticator, noAuthorizator);
+        aliceSession = sessionProvider.createSystemSession(ALICE_ADDRESS);
+        bobSession = sessionProvider.createSystemSession(BOB_ADDRESS);
+
+        mapperFactory = new InMemoryMailboxSessionMapperFactory();
+        DeletedMessageConverter deletedMessageConverter = new DeletedMessageConverter(clock);
+        DeletedMessageVaultHook messageVaultHook = new DeletedMessageVaultHook(sessionProvider, messageVault, deletedMessageConverter, mapperFactory);
+
+        testingData = MessageIdManagerWithPreDeletionHooksTestSystemProvider
+            .createTestingData(sessionProvider, mapperFactory, ImmutableSet.of(messageVaultHook));
+
+        mailboxManager = testingData.getMailboxManager();
+        messageIdManager = testingData.getMessageIdManager();
+
+        internalDate = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss").parse("2019/02/26 14:30:00");
```

Comment:
if we really need a custom date format, could we name it and have a shared constant for it?